nomad/encrypter.go

```diff
@@ -150,22 +215,46 @@ func (e *Encrypter) addCipher(rootKey *structs.RootKey) error {
 		return fmt.Errorf("invalid algorithm %s", rootKey.Meta.Algorithm)
 	}
 
+	privateKey := ed25519.NewKeyFromSeed(rootKey.Key)
+
 	e.lock.Lock()
 	defer e.lock.Unlock()
-	e.ciphers[rootKey.Meta.KeyID] = aead
-	e.keys[rootKey.Meta.KeyID] = rootKey
+	e.keyring[rootKey.Meta.KeyID] = &keyset{
+		rootKey:    rootKey,
+		cipher:     aead,
+		privateKey: privateKey,
+	}
 	return nil
 }
 
 // GetKey retrieves the key material by ID from the keyring
 func (e *Encrypter) GetKey(keyID string) ([]byte, error) {
+	keyset, err := e.keysetByID(keyID)
+	if err != nil {
+		return []byte{}, err
```

Comment:
Just a nitpick. If there's a reason it needs to be an initialized slice best to comment it.

```suggestion
		return nil, err
```